sycl/plugins/unified_runtime/ur/adapters/level_zero/command_buffer.cpp

```diff
@@ -663,6 +663,12 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferEnqueueExp(
     ur_exp_command_buffer_handle_t CommandBuffer, ur_queue_handle_t Queue,
     uint32_t NumEventsInWaitList, const ur_event_handle_t *EventWaitList,
     ur_event_handle_t *Event) {
+  // There are issues with immediate command lists so return an error if the
+  // queue is in that mode.
+  if (Queue->UsingImmCmdLists) {
```

Comment:
We ran into issues when immediate command lists are used because we obtain our prefix and suffix command lists from the queue, so when immediate cmd lists are used that is what the queue will return for those. This was causing hangs when executing the command buffer. We weren't able to track down the problem in time for this release so it seemed more sensible to just error for the moment. 

We are planning to look into the problem in the future so we can support using immediate command lists.

Reply:
thanks @Bensuo for the details. For the prefix and suffix lists, we need also batched (regular) lists. It is likely that what is happening here is that because the immediate lists are submitted immediately, and have better concurrency, that the suffix list reaches the pipeline before the main-body immediate list, and because of that, the suffix list is never signaled, causing the deadlock.

in the command buffers logic, it is better if we us only using batched lists.